independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/scanning/ResteasyReactiveScanner.java

```diff
@@ -262,6 +262,23 @@ public static ResourceScanningResult scanResources(
             }
         }
 
+        // handle abstract classes
+        Set<ClassInfo> abstractScannedResources = scannedResources.values().stream().filter(ClassInfo::isAbstract)
+                .collect(Collectors.toSet());
+        for (ClassInfo abstractScannedResource : abstractScannedResources) {
```

Comment:
Maybe avoid the extra iteration above and call `isAbstract` here?

Reply:
Not sure what you are referring to here

Reply:
You are creating a `Set` above just to iterate again in this loop, I am suggesting merging both loops to a single one